website/blog/2025-07-01-mlflow-prompt-evaluate/index.md

```diff
@@ -0,0 +1,506 @@
+
+```python
+updated_template = """\
+You are an expert at key information extraction and OCR. Extract the questions and answers from the image, where the keys are questions and the values are answers.
+
+
+Question refers to a field in the form that takes in information. Answer refers to the information
+that is filled in the field.
+
+Follow these rules:
+- Only use the information present in the text and do not paraphrase.
+- If the keys have multiple topics, separate them with a slash (/)
+{{ additional_rules }}
+"""
+```
+
+The next step is to register the updated prompt template, load it back later and format it with any additional rule before rerunning the evaluation.
+
+```python
+updated_prompt = mlflow.genai.register_prompt(
+    name="ocr-question-answer",
+    template=updated_template,
+    commit_message="Update commit",
+    tags={
+        "author": "author@example.com",
+        "task": "ocr",
+        "language": "en",
+    },
+)
+
+# Load the updated prompt and format it with additional rules
+
+prompt = mlflow.genai.load_prompt(name_or_uri="ocr-question-answer", version=updated_prompt.version)
+system_prompt = prompt.format(additional_rules="Use exact formatting you see in the form.")
+
+
+results_updated = mlflow.models.evaluate(
+    model=batch_completion,
+    data=pd.DataFrame({"inputs": images, "truth": jsons}),
+    targets="truth",
+    model_type="text",
+    predictions="predictions",
+    extra_metrics=[custom_key_value_accuracy]
+)
+
+print("Custom metric results:", results_updated.metrics)
+eval_table_updated = results_updated.tables["eval_results_table"]
+print("\n Per-row scores:")
+print(eval_table_updated[['key_value_accuracy/score']])
+```
+
+This updated prompt may lead to a marginal improvement in the metric for each image. However, there are still mismatches in values between the ground truth and the LLM response. As showcased above using Prompt Registry, iteratively refining the prompt can address these issues, leading to better alignment with the expected output.
+
+## Conclusion and Next Steps
+
+By leveraging MLflow GenAI capabilities, we efficiently manage prompts and evaluate models for our OCR tool. With all runs, prompts, and metrics logged, we can compare different models or prompt strategies side-by-side in the MLflow UI. This enables data-driven decisions, justifies model selection, and enables both technical and non-technical contributors to collaborate, iterate, and deploy AI solutions confidently.
+
+We can take several directions to further enhance our workflow and outcomes:
+
+**Adopt Managed MLflow:** For enterprise-grade scalability, governance, and zero infrastructure management, try Managed MLflow on Databricks
```

Comment:
This doesn't really address any of the topics in this blog. Let's remove this. It certainly does open up the question about the differences in capability for evaluation between managed MLflow (via databricks-agents evaluation capabilities) and OSS MLflow, a gap that is being closed in the next 2 releases - but let's not focus on that.